pkg/cache/application.go

```diff
@@ -422,7 +422,7 @@ func (app *Application) scheduleTasks(taskScheduleCondition func(t *Task) bool)
 					log.Logger().Warn("init task failed", zap.Error(err))
 				}
 			} else {
-				events.GetRecorder().Event(task.GetTaskPod(), v1.EventTypeWarning, "FailedScheduling", err.Error())
+				events.GetRecorder().Eventf(task.GetTaskPod(), nil, v1.EventTypeWarning, "FailedScheduling", "FailedScheduling", err.Error())
```

Comment:
Minor difference in new API here: https://pkg.go.dev/k8s.io/client-go@v0.20.11/tools/events#EventRecorder

Using `nil` in place of `related` (2nd param) currently since we did not have concept of secondary object in previous Events API. By same logic, we're duplicating `reason` param to be same as `action` since the latter was also not present in previous API. 

Let me know if there's a better structure I can follow to populate these args correctly in `Eventf`.

Reply:
This looks fine to me.